thrust/cmake/ThrustHeaderTesting.cmake

```diff
@@ -148,6 +148,9 @@ function(thrust_add_header_test thrust_target label definitions)
       endif()
     endif()
 
```

Comment:
@alliepiper I have issues with the way the header testing is set up

Everything lands into just one library with one set of compile options.

I had the idea to add a `deprecated` list and then do the same as we du currently but with the additional flag. It was a bit of a hazzle, might revisit